src/Workspaces/Core/Portable/Utilities/IReadOnlyDictionaryExtensions.cs

```diff
@@ -1,20 +1,23 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
 using System.Collections.Generic;
-using Microsoft.CodeAnalysis.Shared.Extensions;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Roslyn.Utilities
 {
     internal static class IReadOnlyDictionaryExtensions
     {
+        [return: MaybeNull]
         public static TValue GetValueOrDefault<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key)
         {
             if (dictionary.TryGetValue(key, out var value))
             {
                 return value;
             }
 
-            return default;
+            return default!;
```

Comment:
Ditto earlier: is this due to the compiler not correctly handling the [MaybeNull]?

Reply:
➡️ Covered by https://github.com/dotnet/roslyn/pull/37512#discussion_r313170343